### ISSUE TITLE
keepassxc-cli incomplete commands/options

### ISSUE THREAD

**Issue by liar666** [keepassxreboot/keepassxc#2220]
Hi,

First of all, MY 1000 THANKS for the tool, I discover more of its power every day ("transform rounds", the cli interface, etc.). It is so great! Also the XC interface is a major improvement over the legacy app (ease of use, configuration, multiple db opened in tabs, etc.)!

I'm trying to convert an old gpg'ed flat-text file I had previously, into a KPxc password database. Since this flat file is quite large, I'm trying to write a script to do the conversion. Unfortunately, I seem to lack a few commands/options in keepassxc-cli to do so.

1. First of all, how would you create a fresh empty db from command line?

2. "keepassxc-cli add/edit" do not seem to provide any access to the "notes" field that exists in the GUI. Why?

Thanks in advance for the help

**Comment by louib** [comment]
@liar666 

> First of all, how would you create a fresh empty db from command line?
That is correct. We would need a `keepassxc-cli create` command to create a fresh database, but it's not there yet.

> "keepassxc-cli add/edit" do not seem to provide any access to the "Notes" field or "Expires" that exists in the GUI. Why? 
No particular reason, I guess nobody needed it until now ;)

> Also @cyphar seems to have added access to additional attributes to "show" in #1289 , but not to "add"/"edit"
I guess that functionality could be added as well!

**Comment by louib** [comment]
Missing `create` function is being tackled in https://github.com/keepassxreboot/keepassxc/pull/2352, I will rename this issue so it's only tracking the missing fields on Add/Edit.

**Comment by liar666** [comment]
Hi,

Thanks for the addition of the "create" function.

Any news for the options (particuliarly the Notes)?

Thanks

**Comment by liar666** [comment]
Hi,

Long story, short: I created this issue because I was searching for a way to move from an old way of managing passes (e.g. plain text file) to keepassxc. But I encontered several problems doing so. One is the present issue.

Finally, I found a work-around! This issue is still relevant (see the note in the end about)

So I share it below for those that would have the same problem.

Here is the solution:
- You can create a keepassxc entry (named e.g. $NAME) with a very long "notes" field, containing the whole list of your previous passwords (e.g. from the content of an old "plain text" file)
- And then you can use:
  DB=/path/to/your.kdbx   # to declare the kdbx file you want to work with
  read -s PASSWORD        # to ask the password from within a script
  echo "$PASSWORD" | keepassxc-cli locate "$DB" "$NAME" # to find the exact path of your entry with the very long "notes" inside the kdbx file
  # Assuming you store the previous result in PATH (I don't do it automatically as the previous command might return several results)
  echo "$PASSWORD" | keepassxc-cli show "$DB" "$PATH" | grep -C 10 "$PATTERN" # to find (based on PATTERN) the "sub"-entry in the "notes" field, that corresponds to the "previously plain text" password you're looking for.



Long story, long:

- For many years, I was using a plain text file, encrypted with gnupg, as my passwords' safe. It was very inconvenient, as I needed to manually decrypt/re-encrypt the file everytime I wanted to add a password.

- A couple of years ago, I've switched to KeepassXC for my new passwords.

- Since then, I was looking for a way to /automatically/ 'convert' my old flat file into a kdbx file that I could merge to my new file. When I say /automatically/, it's because I have >1000 passes in the flat file and I really CAN'T create that much entries manually.

PROBLEM1 (Can't create "notes" field from keepasxc-cli. This issue #2220):
When I discovered keepassxc-cli, I thought about using it to script the conversion (I have markers in my flat file that separate the entries). Unfortunately, many of my 'flat' entries have additional information attached to them that I /need/ to attach in the kdbx file too. I thought of using the "notes" field in the Keepass entries, but this field cannot be acessed from keepassxc-cli for the moment (as can be seen in the relevant issue listed above).

PROBLEM2 (Strange presentation of "search" results #3496)
Since I couldn't create separate entries for my previous passwords, but needed to have all my passwords in the same place (keepassxc), I decided, for the moment, to create a single entry with a very long "notes" field, containing all my previous passwords. This works, but, unfortunately, when I "search" for a "substring" that I know is *only* contained in this "notes" field, keepassxc's GUI returns tens of entries that have nothing to do with my search query... Also, it is impossible to search text within the "notes" field.
So what I did until now is copy/paste the whole "notes" field into a text plain file and search within this text file. This was not convenient AND not secure (whole content is unencrypted in memory + text editor might create hidden backups of the temp file)

(PS: this single entry is not very convinient in KeePassDroid as it takes ages to open it. So the work-around it still not satisfactory... Also I'm not aware of any keepassdroid-cli... so, for the moment, no solution under Android :{ ).

So I was looking for a solution that allowed me:
1. to keep all my passes in a single kdbx file
2. to search for specific sites/passwords
Thus, I combined the two approaches above: single entry with large "notes" field + keepass-cli for lookup/searches. See final work-around on top of this text.


Hope this helps others!

**Comment by droidmonkey** [comment]
You could have simply made the flat file into a csv file and imported that in keepassxc

**Comment by github-throwaway** [comment]
@droidmonkey What's the current status of being able to add an exparation date to an entry via the CLI? I couldn't find it in the docs nor did I see a draft PR for this. :)
